### PR TITLE
Use commands for DHCP test

### PR DESCRIPTION
**Pull request by koln67**
Check if the dhcp IPv4 is found using commands. If any of the cases pass, pass the test.

### CODE REVIEW

File: imagetest/test_suites/network/dhcp_test.go [hunk 1]
```diff
@@ -13,40 +15,151 @@ import (
...
+			return nil
 		}
 	}
-	t.Fatalf("failed finding dhcp client process")
+	return fmt.Errorf("failed finding dhclient process")
+}
+
+// accepts IP addresses in the form of a.b.c.d or a.b.c.d/IPNET
+func validIPOrCIDR(token string) bool {
+	fmt.Printf("token is %s", token)
```
Comment (dorileo) [inline imagetest/test_suites/network/dhcp_test.go:145]: please use `t.Logf()`

File: imagetest/test_suites/network/dhcp_test.go [hunk 2]
```diff
@@ -13,40 +15,151 @@ import (
...
+	IPAddress, _, err := net.ParseCIDR(token)
+	if IPAddress != nil && err == nil {
+		return true
+	}
+
+	return false
+}
+
+func checkCmdExists(cmd string) bool {
+	_, err := exec.LookPath(cmd)
```
Comment (dorileo) [inline imagetest/test_suites/network/dhcp_test.go:159]: if you are checking for cmd (or executables) please also test it's executable.

File: imagetest/test_suites/network/dhcp_test.go (hunk 1 above)
Comment (koln67) [reply imagetest/test_suites/network/dhcp_test.go]: this print statement was used for debugging but is unnecessary in the test, removed.

File: imagetest/test_suites/network/dhcp_test.go (hunk 2 above)
Comment (koln67) [reply imagetest/test_suites/network/dhcp_test.go:158]: https://pkg.go.dev/os/exec#example-LookPath

exec.LookPath() searches for an executable in the path. I believe this checks if the cmd variable is executable, was there a different specific check you had in mind?

Comment (dorileo) [reply imagetest/test_suites/network/dhcp_test.go:158]: `exec.LookPath()` will not check if the file has `+x` mode, you need to check if yourself, you can use the file's `Mode` to check it.

Comment (koln67) [reply imagetest/test_suites/network/dhcp_test.go:158]: these tests run with root/sudo permissions, would this check be necessary?

**Pull request merged**